core/src/main/java/com/exadel/aem/toolkit/core/lists/utils/ListResourceUtils.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceUtil;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.apache.sling.jcr.resource.api.JcrResourceConstants;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.Page;
+import com.adobe.granite.ui.components.ds.ValueMapResource;
+
+import com.exadel.aem.toolkit.core.CoreConstants;
+import com.exadel.aem.toolkit.core.lists.ListConstants;
+import com.exadel.aem.toolkit.core.lists.models.SimpleListItem;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Contains methods for manipulation with List Resource
+ */
+class ListResourceUtils {
+
+    private static final Map<String, Object> LIST_PROPERTIES
+        = Collections.singletonMap(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY, "wcm/foundation/components/responsivegrid");
+
+    private static final List<String> PROPERTIES_TO_IGNORE = Arrays.asList(
+        "jcr:createdBy", "jcr:created", "cq:lastModified", "cq:lastModifiedBy", "jcr:lastModified", "jcr:lastModifiedBy",
+        "cq:lastReplicationAction", "cq:lastReplicatedBy", "cq:lastReplicated"
+    );
+
+    /**
+     * Creates a list resource under given parent
+     * @param resourceResolver Sling {@link ResourceResolver} instance used to create the list
+     * @param parent           List Page that holds list resource
+     * @return list Resource or {@code null} if {@link ResourceResolver} or {@code parent} is null
+     * @throws PersistenceException if list resource cannot be created
+     */
+    static Resource createListResource(ResourceResolver resourceResolver, Page parent) throws PersistenceException {
+        if (resourceResolver == null || parent == null) {
+            return null;
+        }
+
+        return resourceResolver.create(parent.getContentResource(), ListConstants.NN_LIST, LIST_PROPERTIES);
+    }
+
+    /**
+     * Create {@link com.exadel.aem.toolkit.core.lists.models.internal.ListItemModel} resource under {@code parent}
+     * container with given properties.
+     * @param resourceResolver Sling {@link ResourceResolver} instance used to create the list
+     * @param parent           Container for {@code listItem}'s.
+     * @param properties       {@code listItem} properties
+     * @throws PersistenceException if {@code listItem} cannot be created
+     */
+    static void createListItem(ResourceResolver resourceResolver, Resource parent, Map<String, Object> properties) throws PersistenceException {
+        Map<String, Object> withoutSystemProperties = getWithoutSystemProperties(properties);
+        withoutSystemProperties.put(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY, ListConstants.LIST_ITEM_RESOURCE_TYPE);
+        resourceResolver.create(parent, ResourceUtil.createUniqueChildName(parent, CoreConstants.PN_LIST_ITEM), withoutSystemProperties);
+    }
+
+    /**
```

Comment:
To discuss: I find methods in lines 84-116 too granulated and too elementary. I would join them some way. E.g. method `createValueMapResource` is basically a wrapper around a simple constructor.

This can be a matter of taste. This can be even an aftermath of requirements that were in place in some other project (like "you must split code in simple operations; you must cover every operation with a test"). 

I myself don't support this. 
A very split code is harder to read. A method body must tell a "story", like: "we were doing _that_.. and now we got results, and will end with _this_"). But when the code is so split, it starts to remain good old Turbo Basic: goto line XX. Remember what you were doing? Now goto line YY... )

It's all matter of golden mean ("золотая середина") of course ))